testutil/contractmanager/network/network.go

```diff
@@ -0,0 +1,80 @@
+package network
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/cosmos/cosmos-sdk/baseapp"
+	"github.com/cosmos/cosmos-sdk/crypto/hd"
+	"github.com/cosmos/cosmos-sdk/crypto/keyring"
+	servertypes "github.com/cosmos/cosmos-sdk/server/types"
+	"github.com/cosmos/cosmos-sdk/simapp"
+	storetypes "github.com/cosmos/cosmos-sdk/store/types"
+	"github.com/cosmos/cosmos-sdk/testutil/network"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	tmrand "github.com/tendermint/tendermint/libs/rand"
+	tmdb "github.com/tendermint/tm-db"
+
+	"github.com/neutron-org/neutron/app"
+)
+
+type (
+	Network = network.Network
+	Config  = network.Config
+)
+
+// New creates instance with fully configured cosmos network.
+// Accepts optional config, that will be used in place of the DefaultConfig() if provided.
+func New(t *testing.T, configs ...network.Config) *network.Network {
```

Comment:
`configs ...network.Config` — is it a kind of general cosmos pattern? kinda strange taking into account the way we use this New and the returned value. I ask because I can see that we use the same network constructor in ICQ and ICTx modules whereas there are more clean (imo) ways to define optional parameters (e.g. `With...` methods, or several constructors, or even a `nil` value suppport)

this comment is just out of curiosity, an informal one

Reply:
This code is autogenerated by ignite so I guess it was created in this way to be more generic.